client/src/components/stars.jsx

```diff
@@ -24,22 +25,103 @@ const Rating = styled.div`
   width: 132px;
   height: 24px;
   background-image: url("stars.png");
-  background-position: 0 -168px;
+  // Yelp renders their stars by showing a certain part of stars.png based on the rating. 
+  // I created this formula by messing around with different yelp ratings I found for businesses 
+  background-position: 0 ${props => ((-216 - (2* (5 - props.averageScore)) * -24) + 'px')};
+  position: relative; 
+`
+Rating.displayName = "Rating"
+
+const DetailButton = styled.button`
+  border-radius: 3px;
+  border-width: 1px;
+  padding: 3px 5px;
+  margin-left: 10px;
+  border-style: solid;
+  border-color: #ccc;
+  color: #A3A3A3;
+  background-color: #F5F5F5;
+  font-size: .7rem;
+  `
+const Chart = styled.i`
+  padding-right: 5px;
+`
+
+const Link = styled.a`
+  text-decoration: none;
+  color: #A3A3A3;
   position: relative;
+  &:hover #ToolTipNext{
+    visibility: visible;
+  }
+  &:hover{
+    color: black;
+  }
+`
+const NumberOfReviews = styled.span`
 `
+NumberOfReviews.displayName = "NumberOfReviews"
 
+const ToolTipNext = styled.span`
+  visibility: hidden;
+  width: 100px;
+  background-color: black;
+  color: #fff;
+  text-align: center;
+  padding: 5px 0;
+  border-radius: 6px;
+  opacity: 0.9;
 
+  position: absolute;
+  z-index: 1;
```

Comment:
noticed you have been using z-index of 1, 2000, 3000
is that on purpose? or should that one be 1000?

Reply:
well.. in theory i should have 3 layers. 
layer 0, the main content, 
layer 1000, the black modal screen.
layer 2000, the actual modal on top.

I use a z-index of 1 to signify the tooltip belongs on layer 0, but slightly more on top of it. 
if i used a z-index of 2001 (which i haven't used), it would be a tool tip on the modal as it would sit on top of the modal z-index layer

Reply:
awesome. Makes sense man. Thanks for the clarification 👍